### ISSUE TITLE
Spider-Man Remastered ray-tracing artifacts.

### ISSUE THREAD

**Issue by robotman40** [HansKristian-Work/vkd3d-proton#1895]
The game has ray-tracing artifacts that appear on the edges of buildings with ray-tracing enabled. This does not occur when ray tracing is off. The artifacts can be described as being black sharp artifacts and such.

Also, I can confirm it is not a hardware issue since ray-tracing works fine in other games like Forza Horizon 5 with nearly the same performance as Windows.

### Software information
Spider-Man Remastered on Steam

### System information
- GPU: RX 6600 XT
- Driver: Kisak Mesa 24.0
- Wine version: Proton Experimental
- VKD3D-Proton version: Current Proton Experimental vkd3d-proton

**Comment by Blisto91** [comment]
Hit there.

Can you provide screenshots so we know what to look for?
Also a proton log

**Comment by GalacticalEdge** [comment]
> Hit there.
> 
> Can you provide screenshots so we know what to look for? Also a proton log

I will try to get a screenshot later, but it's most noticeable if you go to the tallest building on the south side of the map that is completely glass and run up it vertically.

**Comment by sonic2kk** [comment]
I believe I am encountering the same bug. I realise this issue report is very old, and my hardware and drivers are different (although I am also using an AMD GPU). Let me know if I should open a separate issue. I have not played this game so I don't know too much context about specific levels or settings.

System Specifications:
- Distribution: Arch Linux
- GPU: RX 7900XTX
- Driver: Mesa 24.1.6
- Wine versions tested: Proton 9.0-2, Proton Experimental, GE-Proton9-11, and [a recent Proton-tkg build](https://github.com/Frogging-Family/wine-tkg-git/actions/runs/10523382818) (based on experimental-bleeding-edge)
- vkd3d-proton version: `2.13.0/1.3.270` according to MangoHUD

Arch Linux does not have Mesa 24.2 available yet, but I will also re-test when this version is available.

<hr>

I was able to confirm that the problem is specific to enabling raytracing. Disabling raytracing fixes the issue, so it is easy to workaround.

I don't believe this effect to be specific to GPU load, as it can appear when GPU load is at approximately 40% usage on the Main Menu, approximately 70% usage in vents, and approximately 80-90% usage in the first level open tower area.

The effect is mostly present when there are objects with transparency in a scene, such as glass. However this then bleeds over onto the Spider-Man model as well. Strangely, as far as I can tell, most objects are not affected, just objects with transparency and then Spider-Man.

The problem increases and decreases in severity, mostly when panning the camera.

However this effect is not visible all of the time. On the Main Menu it is not visible, but once you enter the game it is visible (at least at the beginning, where my screenshots are from). However once leaving the game and going back to the Main Menu, the bug _is_ present.

As well as this, in certain areas, the problem is either significantly reduced or not present at all. To my eyes it does not look present but I could be wrong. If exiting in an area where the problem is not present, then the problem will not carry over onto the Main Menu. However if exiting in an area where the problem _is_ present, then it _will_ carry over.

<hr>

Below are some screenshots of the problem at different stages. I tried to capture various levels of intensity, although as the effect does not look as intense in the screenshots.

All screenshots were captured with the game running at close to maximum settings with no upscaling, and no HDR as I do not have a HDR display. While the screenshots were captured at 3840x2160@60Hz, I also tested 1920x1080@60Hz and 1920x1080@165Hz, and they all produced the artifacting. The common denominator appears to be ray tracing.

Screenshots were taken with Steam's screenshot feature using the "Uncompressed" option except where noted otherwise.

### Main Menu - No Artifacting
![1817070_20240823155152_1](https://github.com/user-attachments/assets/220a88a1-328a-45d5-b444-f865920cda4d)

![1817070_20240823155156_1](https://github.com/user-attachments/assets/cbc4433f-5560-445f-87b9-977513eacba3)

![1817070_20240823155207_1](https://github.com/user-attachments/assets/adfb21e1-e073-4930-8d91-65338639c42d)

### Level Example - Artifacting Present
This is the area in the beginning of the game where the problem is most prevalent. Many apologies but I had to compress these screenshots as they were slightly too large to upload uncompressed.

![1817070_20240823153538_1](https://github.com/user-attachments/assets/98b5eb67-fc37-4edb-81e4-ba42d5db5bfe)

![1817070_20240823153544_1](https://github.com/user-attachments/assets/ec0d8b1d-d058-4b6b-9918-f2905d793f68)

### Main Menu After Experiencing Problem - Artifacting Carried Over
![1817070_20240823154514_1](https://github.com/user-attachments/assets/33e25288-cdf0-4153-83b6-5f8f593c3ee3)

### Enclosed Area - Issue Not Present
If exiting from this area, the artifacting above is not present on the Main Menu.

![1817070_20240823154314_1](https://github.com/user-attachments/assets/77d50036-742f-4fd1-ae02-367232d13ab2)

### Less Enclosed Area - Artifacting Present But Not Captured
The issue also occurs in the area shown in this screenshot, but I could not capture it. The issue is much more reduced than the main screenshot. I was unable to get this to appear on a screenshot, and I could not return to this area for an additional screenshot. However I am attaching anyway in case the effects present in this area give hints as to the cause of the artifacting.

![1817070_20240823153612_1](https://github.com/user-attachments/assets/9d5f9b8c-f38c-4354-95bc-93f87fa798e4)

<hr>

@robotman40 please let me know if these match the artifacts you are describing. If they are not, many apologies and I can open a separate issue.

**Comment by sonic2kk** [comment]
I discovered that this appears to be similar to #1191.

**Comment by Blisto91** [comment]
I've never been able to reproduce either issue after a lot of trying.
But will give it a shot again later.

**Comment by mbriar** [comment]
@sonic2kk are you sure you're not using amdvlk, post `vulkaninfo | grep driver`. Sorry, but on arch linux i always feel like asking.

**Comment by sonic2kk** [comment]
Thanks for asking, but I am sure. I don't have amdvlk as far as I'm aware.

**Comment by sonic2kk** [comment]
I updated to Mesa 24.2.1 today and wanted to update that this issue is still present. If it is a Mesa issue I can report it upstream.

**Comment by K0bin** [comment]
> I discovered that this appears to be similar to https://github.com/HansKristian-Work/vkd3d-proton/issues/1191.

That happens without ray tracing though.

**Comment by sonic2kk** [comment]
Yes, I was just noting that the effect looks similar. The problem in this issue, at least in the way I am experiencing it (and the way I understand OP to be experiencing it), only happens with raytracing, or at the very least I have been unable to reproduce it in any other way.

**Comment by robotman40** [comment]
If I remember correctly, this was fixed in vkd3d-proton 3.0. So I'll close this issue.
